File: prime_expanding_memory.go
package main

import "fmt"

func main() {

  var slice_primes []int;

  for i := 2; i < 20000000; i++ {
    is_prime := true;
    for j := 0; j < len(slice_primes) && slice_primes[j] * slice_primes[j] <= i; j++ {
      if i % slice_primes[j] == 0 {
        is_prime = false;
        break;
      }
    }
    if is_prime {
      slice_primes = append(slice_primes, i);
    }
  }
  fmt.Println(len(slice_primes));
  bufio.NewReader(os.Stdin).ReadBytes('\n')

}

//Compiled: go build -ldflags="-s -w"
//Run Time: 6.2 sec
//Memory before: 1,532 K
//Memory After: Ranges between 29,000 K and 45,000 k

File: prime_expanding_memory.rs
fn main() {

  let mut vec_primes: Vec<u64> = Vec::with_capacity(20000000);
  for i in 2..20000000 {
    if vec_primes.iter().take_while(|&p| p * p <= i).all(|&p| i % p != 0) {
      vec_primes.push(i);
    }
  }
  println!("{}", vec_primes.len());

}

//Compiled: cargo build --release
//6.0 sec
//Memory Before: 564 K
//Memory After: 10,512 K
